extension/src/state_aggregate/rollup.rs

```diff
@@ -187,8 +199,8 @@ impl OwnedCompactStateAgg {
 
             first_time: earlier.first_time,
             last_time: later.last_time,
-            first_state: earlier.first_state,
-            last_state: added_entries + later.last_state,
+            first_state: earlier.first_state, // indexes into earlier durations are same for merged_durations
+            last_state: merged_last_state,
```

Comment:
It would be good to add a test case that fails with the old code, but passes with the error fixed.

Reply:
I expanded a test I added in this PR so that:
- removing the aggregate sorting results in `merges_compact_aggs_correctly` failing
- removing the fix to `last_state` results in the state_agg doctest failing